test/unit/three_scale/analytics/user_tracking_test.rb

```diff
@@ -8,125 +8,155 @@ def setup
+    end
 
-    tracking = UserTracking.new(@user)
-    assert tracking.track('Event')
+    def test_group
+      tracking = UserTracking.new(nil)
+      refute tracking.group
 
-    @user.account.provider = false
-    refute tracking.track('Event')
-  end
+      tracking = UserTracking.new(@user)
+      assert tracking.group
 
-  def test_group
-    tracking = UserTracking.new(nil)
-    refute tracking.group
+      @user.account.provider = false
+      refute tracking.group
+    end
 
-    tracking = UserTracking.new(@user)
-    assert tracking.group
+    def test_group_traits
+      tracking = UserTracking.new(nil)
+      assert_equal Hash.new, tracking.group_traits
 
-    @user.account.provider = false
-    refute tracking.group
-  end
+      tracking = UserTracking.new(@user)
+      @user.account.stubs(bought_plan: Plan.new(name: 'fooplan', cost_per_month: 42), state: 'suspended')
 
-  def test_group_traits
-    tracking = UserTracking.new(nil)
-    assert_equal Hash.new, tracking.group_traits
+      assert_equal({name: @user.account.name,
+                    plan: 'fooplan',
+                    monthly_spend: 42.0,
+                    license_mrr: 42.0,
+                    state: 'suspended',
+                   }, tracking.group_traits)
+    end
 
-    tracking = UserTracking.new(@user)
-    @user.account.stubs(bought_plan: Plan.new(name: 'fooplan', cost_per_month: 42), state: 'suspended')
+    def test_can_send?
+      refute UserTracking.new(nil).can_send?
+      assert UserTracking.new(@user).can_send?
+    end
 
-    assert_equal({name: @user.account.name,
-                  plan: 'fooplan',
-                  monthly_spend: 42.0,
-                  license_mrr: 42.0,
-                  state: 'suspended',
-                 }, tracking.group_traits)
-  end
+    def test_only_track_providers
+      @user.account.provider = false
+      refute UserTracking.new(@user).can_send?
+    end
 
-  def test_can_send?
-    refute UserTracking.new(nil).can_send?
-    assert UserTracking.new(@user).can_send?
+    def test_do_not_track_impersonation_admins
+      @user.username = ThreeScale.config.impersonation_admin['username']
+      refute UserTracking.new(@user).can_send?
+    end
   end
 
-  def test_only_track_providers
-    @user.account.provider = false
-    refute UserTracking.new(@user).can_send?
-  end
+  class SegmentDisabled < UserTrackingTest
+    def setup
+      super
+      Features::SegmentConfig.stubs(enabled?: false)
+    end
+
+    def test_does_not_track
+      ThreeScale::Analytics::UserTracking::TrackingAdapter::NullAdapter.any_instance.expects(:track).once
+      # ThreeScale::Analytics::UserTracking::TrackingAdapter::NullAdapter.any_instance.expects(:track).once
+      UserTracking.new(@user).track('foo')
+    end
 
-  def test_do_not_track_impersonation_admins
-    @user.username = ThreeScale.config.impersonation_admin['username']
-    refute UserTracking.new(@user).can_send?
+    # def test_does_not_identify
```

Comment:
What about those tests? Are they needed? If so will you fix them or do you think it is unneeded then remove>

Reply:
Me & @Martouta are playing with some tests.. We will fix it 😄 ..

Reply:
will add a wip tag on the PR